Source/WebCore/rendering/GridMasonryLayout.cpp

```diff
@@ -175,8 +201,9 @@ void GridMasonryLayout::setItemGridAxisContainingBlockToGridArea(const GridTrack
     gridItem.setChildNeedsLayout(MarkOnlyThis);
 }
 
-void GridMasonryLayout::insertIntoGridAndLayoutItem(const GridTrackSizingAlgorithm& algorithm, RenderBox& gridItem, const GridArea& area)
+void GridMasonryLayout::insertIntoGridAndLayoutItem(const GridTrackSizingAlgorithm& algorithm, RenderBox& gridItem, const GridArea& area, GridMasonryLayout::OverrideLogicalWidth overrideLogicalWidthType)
 {
+    overrideLogicalWidth(gridItem, overrideLogicalWidthType);
```

Comment:
imo it's not great to have a generic method like this that takes in a renderer and mutates it in such a way. I feel like it makes it easy to misuse. Why not just have this method compute a size and shift the responsibility to the caller to figure out what to do with it since they probably have more context as to what's going on.

Reply:
That seems fine will update this to have the above function do the setting of logical widths.